crates/telemetry/src/alert_in_dev.rs

```diff
@@ -0,0 +1,41 @@
+//! Provides a way for warnings to be force-raised in the
+//! `spin up` environment even if RUST_LOG is not set to warn.
+//! This is useful for things that are not errors but where we
+//! want application developers to know they have a problem.
+
+use tracing::{Event, Subscriber};
+use tracing_subscriber::Layer;
+
+const ALERT_IN_DEV_TAG: &str = "alert_in_dev";
+
+/// A layer which prints a terminal warning (using [terminal::warn!]) if
+/// a trace event contains the tag "alert_in_dev" (with any value).
+pub(crate) fn alert_in_dev_layer<S: Subscriber>() -> impl Layer<S> {
+    CommandLineAlertingLayer
+}
+
+pub struct CommandLineAlertingLayer;
+
+impl<S: Subscriber> Layer<S> for CommandLineAlertingLayer {
+    fn on_event(&self, event: &Event<'_>, _ctx: tracing_subscriber::layer::Context<'_, S>) {
+        let meta = event.metadata().fields();
+        if meta.field(ALERT_IN_DEV_TAG).is_some() {
+            warn(event);
+        }
+    }
+}
```

Comment:
I think tracing "wants" you to do conditional logging with a [layer filter](https://docs.rs/tracing-subscriber/0.3.19/tracing_subscriber/layer/index.html#per-layer-filtering), something like:

```suggestion
    CommandLineAlertingLayer
    	.with_filter(filter_fn(|meta| {
    	    meta.field(ALERT_IN_DEV_TAG).is_some()
        }))

}

pub struct CommandLineAlertingLayer;

impl<S: Subscriber> Layer<S> for CommandLineAlertingLayer {
    fn on_event(&self, event: &Event<'_>, _ctx: tracing_subscriber::layer::Context<'_, S>) {
        warn(event);
    }
}
```

The difference is that the bare layer will - by default - register "interest" in _all_ events, preventing tracing from short-circuiting processing of events that no subscriber cares about.